### PR TITLE
Downstream changes

### PR DESCRIPTION
**Pull request by nikitalita**
This adds the following commits from @eckserah , @fireundubh , and @infomaniac50 's forks:

- eckserah: Update to boost 1.71 and some fixes 
  - Fixes an issue with JMPT and JMPF where Integers could be used for the first arguement but wasnt being processed.
  - Add the new opcode for fallout76 (no idea how it works just yet)
- eckserah: Tabular changes
  - (just formatting changes)
- fireundubh: Fixed capitalization on some tokens
  - Fixed capitalization of some tokens
  - Replaced tabs with spaces (This fixes the previous eckserah commit which used tabs instead of spaces)
  - Removed header and tree dump
  - Removed rebuild log
  - Updated Boost to 1.74
- infomaniac50: Build with vcpkg
- infomaniac50: Bump version 
  - (bumps version to v1.0.8)

### CODE REVIEW

File: Pex/Instruction.cpp [hunk 1]
```diff
@@ -72,7 +72,8 @@ static const OpCodeInfo OPCODES[Pex::OpCode::MAX_OPCODE] = {
     {"array_insert", 3, false},
     {"array_removelast", 1, false},
     {"array_remove", 3, false},
-    {"array_clear", 1, false}
+    {"array_clear", 1, false},
+	{"array_getallmatchingstructs", 6, false},
```
Comment (Orvid) [inline Pex/Instruction.cpp:76]: Here and in `Pex/Instruction.hpp:61` it looks like tabs have been added for indents and didn't get cleaned up as part of the prior tab cleanup.

File: Decompiler/PscDecompiler.hpp [hunk 2]
```diff
@@ -11,7 +11,7 @@
 
 #include "Node/Base.hpp"
 
-//#define TRACE_DECOMPILATION
+#define TRACE_DECOMPILATION
```
Comment (Orvid) [inline Decompiler/PscDecompiler.hpp:14]: Trace should likely be off by default, possibly shifting it to a runtime flag in the future

File: Decompiler/PscDecompiler.cpp [hunk 3]
```diff
@@ -1379,11 +1379,6 @@ void Decompiler::PscDecompiler::dumpBlock(size_t startBlock, size_t endBlock)
             }
             m_Log << '\n';
         }
-        DumpTree tree([&] (std::ostream& line)
```
Comment (Orvid) [inline Decompiler/PscDecompiler.cpp:1382]: The removed parts of the trace logging should probably be added back in future commits, such as what was removed here.

File: Decompiler/PscDecompiler.cpp [hunk 4]
```diff
@@ -1287,13 +1294,6 @@ void Decompiler::PscDecompiler::cleanUpTree(Node::BasePtr program)
  */
 void Decompiler::PscDecompiler::generateCode(Node::BasePtr program)
 {
-#ifdef TRACE_DECOMPILATION
```
Comment (Orvid) [inline Decompiler/PscDecompiler.cpp:1290]: The removed parts of the trace logging should probably be added back in future commits, such as what was removed here.

File: Decompiler/PscDecompiler.cpp [hunk 5]
```diff
@@ -61,10 +56,6 @@ Decompiler::PscDecompiler::PscDecompiler(const Pex::Function &function, const Pe
     m_Object(object),
     m_CommentAsm(commentAsm)
 {
-#ifdef TRACE_DECOMPILATION
```
Comment (Orvid) [inline Decompiler/PscDecompiler.cpp:64]: The removed parts of the trace logging should probably be added back in future commits, such as what was removed here.

File: Decompiler/PscDecompiler.cpp [hunk 6]
```diff
@@ -1,18 +1,13 @@
 #include "PscDecompiler.hpp"
 
-#include <map>
 #include <cassert>
 #include <cstdint>
 #include <iostream>
-#include <sstream>
```
Comment (Orvid) [inline Decompiler/PscDecompiler.cpp:7]: The `sstream` include removal here, and possible a few of the other includes that were removed were only needed for the `TRACE_DECOMPILATION` code, so should be added back at the same time.

File: Decompiler/PscCoder.cpp [hunk 7]
```diff
@@ -56,32 +55,6 @@ Decompiler::PscCoder &Decompiler::PscCoder::outputAsmComment(bool commentAsm)
     return *this;
 }
 
-/**
- * @brief Write the content of the PEX header as a block comment.
- * @param pex Binary to decompile.
- */
-void Decompiler::PscCoder::writeHeader(const Pex::Binary &pex)
```
Comment (Orvid) [inline Decompiler/PscCoder.cpp:63]: There was a specific question in the discord about why this was removed in one of the prior commits, and it probably makes sense to add it back, though the version info will need to be updated for the new champollion version number.

File: Decompiler/DumpTree.hpp [hunk 8]
```diff
@@ -161,7 +161,7 @@ class DumpTree : public Node::VisitorBase
     {
         enter("ArrayCreate", node);
         m_Callback(indent() << "Type:" << node->getType());
-        m_Callback(indent() << "Size:" << node->getSize());
+        //m_Callback(indent() << "Size:" << node->getSize());
```
Comment (Orvid) [inline Decompiler/DumpTree.hpp:164]: Any idea why this got commented out?

**Review by Orvid — APPROVED** [review]
A couple of minor comments, though feel free to land as-is.

**Comment by nikitalita** [comment]
I have these fixed in the next PR, going to fix them there instead of attempting to rebase

**Pull request merged**